workspaces/global-header/plugins/global-header/src/components/SearchComponent/SearchBar.tsx

```diff
@@ -0,0 +1,151 @@
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import Autocomplete from '@mui/material/Autocomplete';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import SearchIcon from '@mui/icons-material/Search';
+import { highlightMatch } from '../../utils/stringUtils';
+import styles from './SearchBar.module.css';
+import { useNavigate } from 'react-router-dom';
+
+interface SearchBarProps {
+  query: SearchResultProps['query'];
+  setSearchTerm: (term: string) => void;
+}
+export const SearchBar = (props: SearchBarProps) => {
+  const { query, setSearchTerm } = props;
+  const navigate = useNavigate();
+
+  return (
+    <SearchResultState {...props}>
+      {({ loading, error, value }) => {
+        const results = query?.term ? value?.results ?? [] : [];
+        let options: string[] = [];
+        if (results.length > 0) {
+          options = [
+            ...results.slice(0, 5).map(result => result.document.title),
+            `${query?.term}`,
+          ];
+        } else if (query?.term) {
+          options = ['No results found'];
+        }
+
+        return (
+          <Autocomplete
+            freeSolo
+            options={options}
+            loading={loading}
+            getOptionLabel={option => option ?? ''}
+            onInputChange={(_, inputValue) => setSearchTerm(inputValue)}
+            sx={{ width: '100%' }}
+            filterOptions={x => x}
+            getOptionDisabled={option => option === 'No results found'}
+            onKeyDown={event => {
+              if (event.key === 'Enter') {
+                event.preventDefault();
+                if (query?.term) {
+                  navigate(`/search?query=${query?.term}`);
+                }
+              }
+            }}
+            renderInput={params => (
+              <TextField
+                {...params}
+                placeholder="Search..."
+                variant="standard"
+                error={!!error}
+                helperText={error ? 'Error fetching results' : ''}
+                InputProps={{
+                  ...params.InputProps,
+                  disableUnderline: true,
+                  startAdornment: (
+                    <InputAdornment position="start">
+                      <SearchIcon style={{ color: '#fff' }} />
+                    </InputAdornment>
+                  ),
+                }}
+                sx={{
+                  pt: '0.5rem',
+                  input: { color: '#fff' },
+                  button: { color: '#fff' },
+                }}
+              />
+            )}
+            renderOption={(renderProps, option, { index }) => {
+              if (option === query?.term && index === options.length - 1) {
+                return (
+                  <Box key="all-results" id="all-results">
+                    <Divider sx={{ my: 0.5 }} />
+                    <Link to={`/search?query=${query?.term}`} underline="none">
```

Comment:
If you don't escape query parameters, you search can't include URL characters like `&`:

```suggestion
                    <Link to={`/search?query=${encodeURIComponent(query?.term)}`} underline="none">
```

Since this happen two times now, maybe you can create a small inline function createSearchLink, or something similar.